Cargo.toml

```diff
@@ -41,13 +41,13 @@ cfg-if = "1.0"
 thiserror = "1.0"
 bzip2 = { version = "0.4.4", optional = true }
 flate2 = { version = "1.0", optional = true }
-liblzma = { version = "0.3.1", optional = true }
-bgzip = { version = "0.2.2", optional = true }
-zstd = { version = "0.12.2", optional = true }
+xz2 = { version = "0.1", optional = true }
+bgzip = { version = "0.3", optional = true }
+zstd = { version = "0.13", optional = true }
 
 [dev-dependencies]
 tempfile = "3"
-criterion = "0.3"
+criterion = "0.5"
```

Comment:
This one brings our MSRV to `1.74`. Maybe we can change the MSRV check from `cargo test` to `cargo build`, then we can use newer dev dependencies without issues.

Reply:
In fact we didn't need criterion 0.5 so we can go back to criterion 0.3